src/client/shader.cpp

```diff
@@ -605,6 +615,13 @@ ShaderInfo generate_shader(const std::string &name, u8 material_type, u8 drawtyp
 	// Create shaders header
 	std::string shaders_header = "#version 120\n";
 
+	// For renderers that should use discard instead of GL_ALPHA_TEST
+	const char* gl_renderer = (const char*)glGetString(GL_RENDERER);
+	printf("GL_RENDERER: %s\n", gl_renderer);
+	if (strstr(gl_renderer, "GC7000")) {
+		shaders_header += "#define USE_DISCARD\n";
+	}
```

Comment:
you can additionally check if `materialtype == TILE_MATERIAL_LIQUID_OPAQUE`, `TILE_MATERIAL_OPAQUE` or `TILE_MATERIAL_WAVING_LIQUID_OPAQUE` here.

Reply:
Thanks! Ah, and forgot a printf in there, sorry. Also, glGetString seems to cause trouble with the Windows build?

Reply:
Also the windows build is failing since Minetest usually does not directly call into OpenGL and GL linking is only present on Linux.

You could either put this code into an `#ifdef __unix__` or `#if ENABLE_GLES` if you don't expect users to actually use desktop GL.

Reply:
I'm trying to understand why you chose these 3 material types. Could you explain? Do the other types never have textures with transparent pixels?

Reply:
Added check for __unix__.